### PR TITLE
kubelet test enhancements

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This makes maintenance of the kubelet tests a bit easier. Relevant for https://github.com/kubernetes/kubernetes/pull/125488.

#### Special notes for your reviewer:

~~@SergeyKanzhelev~~ @ffromani  agreed on Slack that replacing the manually written deep comparison code with cmp.Diff is desirable. The second commit implements.

One of the existing tests is either broken or there is a bug in the code. I don't have time to investigate and thus commented out that test case.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @klueska @SergeyKanzhelev

### CODE REVIEW

**Comment by pohly** [comment]
The error in "pull-kubernetes-kind-classic-dra" can be ignored. This PR isn't ready for that yet (needs new feature gate from https://github.com/kubernetes/kubernetes/pull/125488).

File: pkg/kubelet/apis/podresources/server_v1_test.go [hunk 1]
```diff
@@ -674,14 +679,14 @@ func TestAllocatableResources(t *testing.T) {
...
-			},
-		},
+		// {
+		// 	desc:       "no devices, no CPUs, all memory",
+		// 	allCPUs:    []int64{},
+		// 	allDevices: []*podresourcesapi.ContainerDevices{},
+		// 	expectedAllocatableResourcesResponse: &podresourcesapi.AllocatableResourcesResponse{
+		// 		Memory: allMemory,
+		// 	},
+		// },
```
Comment (ffromani) [inline pkg/kubelet/apis/podresources/server_v1_test.go:689]: the fix for the memory part is simple:
```
-               // {
-               //      desc:       "no devices, no CPUs, all memory",
-               //      allCPUs:    []int64{},
-               //      allDevices: []*podresourcesapi.ContainerDevices{},
-               //      expectedAllocatableResourcesResponse: &podresourcesapi.AllocatableResourcesResponse{
-               //              Memory: allMemory,
-               //      },
-               // },
+               {
+                       desc:       "no devices, no CPUs, all memory",
+                       allCPUs:    []int64{},
+                       allDevices: []*podresourcesapi.ContainerDevices{},
+                       allMemory:  allMemory,
+                       expectedAllocatableResourcesResponse: &podresourcesapi.AllocatableResourcesResponse{
+                               Memory: allMemory,
+                       },
+               },
```

equivalently, vs current master (without this PR):
```
$ git diff
diff --git a/pkg/kubelet/apis/podresources/server_v1_test.go b/pkg/kubelet/apis/podresources/server_v1_test.go
index aaa560fc4f0..17ae5c6c2a5 100644
--- a/pkg/kubelet/apis/podresources/server_v1_test.go
+++ b/pkg/kubelet/apis/podresources/server_v1_test.go
@@ -678,6 +678,7 @@ func TestAllocatableResources(t *testing.T) {
                        desc:       "no devices, no CPUs, all memory",
                        allCPUs:    []int64{},
                        allDevices: []*podresourcesapi.ContainerDevices{},
+                       allMemory:  allMemory,
                        expectedAllocatableResourcesResponse: &podresourcesapi.AllocatableResourcesResponse{
                                Memory: allMemory,
                        },
```

**Review by ffromani — COMMENTED** [review]
LGTM once comments are addressed to restore the broken tests. I'd be also fine with a followup PR to fix the commented test.

**Comment by pohly** [comment]
@ffromani: I re-enabled and fixed the test. Thanks for checking.

**Review by ffromani — COMMENTED** [review]
/lgtm

thanks @pohly 
I like the cleanup and I like more the new error. Appreciated!

**Comment by bart0sh** [comment]
/triage accepted
/priority important-soon

/assign @klueska @SergeyKanzhelev @dims 
for approval

**Review by mrunalp — APPROVED** [review]

**Pull request merged**